bolt/src/main/java/com/slack/api/bolt/service/builtin/AmazonS3InstallationService.java

```diff
@@ -14,25 +15,49 @@
 import com.slack.api.bolt.service.InstallationService;
 import com.slack.api.bolt.util.JsonOps;
 import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 import java.io.IOException;
 import java.util.Optional;
 
 @Slf4j
 public class AmazonS3InstallationService implements InstallationService {
 
+    @Nullable
+    private String region;
+    private String accessKey;
+    private String secretKey;
+
     private final String bucketName;
     private boolean historicalDataEnabled;
 
+    public AmazonS3InstallationService(@Nullable String region, @NotNull String accessKey, @NotNull String secretKey, @NotNull String bucketName) {
```

Comment:
Having four string arguments is a very easy-to-use interface. more importantly this is not frexible enough for future enhancement. I think receiving a single object representing crendential inputs (either AWS SDK's class or a custom class with builder interface) + bucketName works better for this use case.

Reply:
OK, depends on you. Just provide a constructor for user to prepare necessary values/keys. Rather than deep digging the system looking for variables. 
This also eases testing / mocking.

Reply:
Point : Let user control what is inputed .